documentation/bolt_types_reference.md

```diff
@@ -0,0 +1,319 @@
+# Bolt data types
+
+## `ApplyResult`
```

Comment:
Defer to @hestonhoffman on this, but I think we should dedup this documentation from https://puppet.com/docs/bolt/latest/writing_plans.html#handling-plan-function-results and link to it from there. Also need to update sentence here: https://puppet.com/docs/bolt/latest/applying_manifest_blocks.html#return-value-of-apply-action

Reply:
I removed the bits from that doc that referenced data types and linked to this doc. Also fixed that link in the second doc